src/lib/swapper/swappers/CowSwapper/utils/helpers/helpers.ts

```diff
@@ -158,19 +162,29 @@ type GetValuesFromQuoteResponseArgs = {
   buyAsset: Asset
   sellAsset: Asset
   response: CowSwapQuoteResponse
+  affiliateBps: string
 }
 
 export const getValuesFromQuoteResponse = ({
   buyAsset,
   sellAsset,
   response,
+  affiliateBps,
 }: GetValuesFromQuoteResponseArgs) => {
   const {
     sellAmount: sellAmountAfterFeesCryptoBaseUnit,
     feeAmount: feeAmountInSellTokenCryptoBaseUnit,
-    buyAmount: buyAmountAfterFeesCryptoBaseUnit,
+    buyAmount,
   } = response.quote
 
+  const hasAffiliateFee = bn(affiliateBps).gt(0)
+  // Remove affiliate fees off the buyAmount to get the amount after affiliate fees, but before slippage bips
```

Comment:
Another needed hack to circumvent the API not serving us those calculations - in other words, CoW API doesn't do this for us so this is the *actual* buyAmount